src/System.CommandLine/Platform.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace System.CommandLine
+{
+    internal static class Platform
+    {
+        private static bool? _isWasm;
+        public static bool IsWasm
```

Comment:
Maybe call this something like `IsConsoleRedirectionSupported` or something? This isn't definitive proof that we're running under WASM, although I'm not aware of other cases where this would happen.

Reply:
kind of agree but note that the not supported operation ahs nothing to do with redirection per se, that seems to be support, is to call IsOutputRedirected, or IsErrorRedirected or IsInputRedirected that is not supported.